pilot/pkg/credentials/kube/secrets.go

```diff
@@ -57,6 +57,15 @@ const (
 	TLSSecretCaCert = "ca.crt"
 	// The ID/name for the CRL in kubernetes tls secret.
 	TLSSecretCrl = "ca.crl"
+
+	// The ID/name for filename in kubernetes generic secret.
+	DataSourceFileName = "filename"
+	// The ID/name for inline_bytes in kubernetes generic secret.
+	DataSourceInlineBytes = "inline_bytes"
+	// The ID/name for inline_string in generic secret.
+	DataSourceInlineString = "inline_string"
+	// The ID/name for environment_variable in generic secret.
+	DataSourceEnvironmentVariable = "environment_variable"
```

Comment:
These seems not a standard key

Reply:
Referred from here: https://github.com/envoyproxy/envoy/blob/main/api/envoy/config/core/v3/base.proto#L394

Reply:
Yes, I know that is envoy api field. It may conflict with uer settings.

BTW, for inline_string and inline_bytes, we should only have one. They can convert to each other

Reply:
> Yes, I know that is envoy api field. It may conflict with uer settings.

Oh, can you help me with an example? If regarding TLS, I assume we would check the TLS flow first always (as proposed) and then generic secrets presence can be checked if the cert is not found.

> BTW, for inline_string and inline_bytes, we should only have one. They can convert to each other
ack, added both for completeness sake, will remove `inline_string` once ready for review.